repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -55,518 +54,373 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
-              case Failure(e: UnauthorizedEndpointException) => Reject(SC_FORBIDDEN, Some(e.getMessage))
-              case Failure(e: OverLimitException) =>
-                response.addHeader(HttpHeaders.RETRY_AFTER, e.retryAfter)
-                if (isSelfValidating) {
-                  Reject(e.statusCode, Some(e.getMessage))
-                } else {
-                  Reject(SC_SERVICE_UNAVAILABLE, Some(e.getMessage))
-                }
-              case Failure(e) if e.getCause.isInstanceOf[AkkaServiceClientException] && e.getCause.getCause.isInstanceOf[TimeoutException] =>
-                Reject(SC_GATEWAY_TIMEOUT, Some(s"Call timed out: ${e.getMessage}"))
-              case Failure(_: AdminTokenUnauthorizedException) if isSelfValidating =>
-                Reject(SC_UNAUTHORIZED, Some("Token unauthorized"))
-              case Failure(e) => Reject(SC_INTERNAL_SERVER_ERROR, Some(e.getMessage))
-            }
-          }
+  override def doWork(servletRequest: HttpServletRequest, servletResponse: HttpServletResponse, chain: FilterChain): Unit = {
+    val response = new HttpServletResponseWrapper(servletResponse, MUTABLE, PASSTHROUGH)
 
-        filterResult match {
-          case Pass =>
-            logger.trace("Processing completed, passing to next filter or service")
-            addIdentityStatusHeader(confirmed = true)
-            chain.doFilter(request, response)
-          case Reject(statusCode, message) =>
-            Option(config.getDelegating) match {
-              case Some(delegating) =>
-                logger.debug(s"Delegating with status $statusCode caused by: ${message.getOrElse("unspecified")}")
-
-                val delegationHeaders = buildDelegationHeaders(statusCode,
-                  "keystone-v2",
-                  message.getOrElse("Failure in the Keystone v2 filter").replace("\n", " "),
-                  delegating.getQuality)
-
-                addIdentityStatusHeader(confirmed = false)
-                delegationHeaders foreach { case (key, values) =>
-                  values foreach { value =>
-                    request.addHeader(key, value)
-                  }
-                }
+    super.doWork(servletRequest, response, chain)
 
-                chain.doFilter(request, response)
+    val keystoneAuthenticateHeader = s"Keystone uri=${configuration.getIdentityService.getUri}"
 
-                logger.trace(s"Processing response with status code: $statusCode")
+    response.uncommit()
+    if (response.getStatus == SC_UNAUTHORIZED) {
+      response.addHeader(HttpHeaders.WWW_AUTHENTICATE, keystoneAuthenticateHeader)
+    }
+    response.commitToResponse()
+  }
 
-              case None =>
-                logger.debug(s"Rejecting with status $statusCode")
+  override def doAuth(request: HttpServletRequestWrapper): Try[Unit.type] = {
+    /**
+      * STATIC REFERENCE TO CONFIG
+      */
+    val config = configuration
```

Comment:
Hm, keeping this line here kind of breaks the idea of the static reference. The current configuration object is already captured in the `doWork` method of the abstract filter. Capturing it again here using the global variable means that it could have changed between when it was captured in `doWork` and here. I suppose the thing to do here, if we do want to make sure that the configuration doesn't change under us, is to pass the config reference from `doWork` into `doAuth`.

Reply:
I was never a fan of this being here. I was just preserving what it use to do. I have no qualms about just cutting it entirely.

Reply:
I'd say go for it, then. Using the global variable holding the configuration simplifies everything a little bit, and it means that methods can be de-nested (un-nested? whatever) without having to pass the configuration around all the time. That makes things easier to read, I think, and easier to unit test.

Besides, few if any of our other filters keep a static reference, and I can't think of a time where the config changing during processing has been an issue.